### PR TITLE
charm: open port NATS is listening on

### PR DESCRIPTION
**Pull request by morphis**
The operator framework currently doesn't not support opening or closing
ports the charm exposes. Due to this we have to manually invoke the
open-port and close-port hook tools. The implementation also keeps track
of changing ports.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -234,9 +235,16 @@ def reconfigure_nats(self):
             self.NATS_SERVER_CONFIG_PATH.write_text(rendered_content)
             if self.state.is_started:
                 subprocess.check_call(['systemctl', 'restart', self.NATS_SERVICE])
-        self.unit.status = ActiveStatus()
         self.client.expose_nats(auth_token=self.state.auth_token)
 
+        port = '{}/tcp'.format(self.model.config['client-port'])
```
Comment (aguadoenzo) [inline src/charm.py:240]: Doesn't this risk an Exception if `client-port` is unset in the configuration ?

**Review by aguadoenzo — CHANGES_REQUESTED** [review]

File: src/charm.py (hunk 1 above)
Comment (morphis) [reply src/charm.py:240]: It should be always set, shouldn't it?

Comment (aguadoenzo) [reply src/charm.py:240]: I'm not 100% sure how the operator framework handles this, but what would be the behavior if we run `juju config nats client-port=` ? Also we should probably check that the port is a valid one before trying to open some nonsense port like `None/tcp` or the likes

Comment (morphis) [reply src/charm.py:240]: A `juju config client-port=` should still not remove the config key from the map. Checking for an empty string or a none value is indeed something we should do.

**Review by aguadoenzo — APPROVED** [review]
LTGM

**Pull request merged**